src/python/grpcio_health_checking/grpc_health/v1/health.py

```diff
@@ -59,20 +60,35 @@ def close(self):
             self._condition.notify()
 
 
+def _watcher_to_on_next_callback_adapter(watcher):
+
+    def on_next_callback(response):
+        if response is None:
+            watcher.close()
+        else:
+            watcher.add(response)
+
+    return on_next_callback
+
+
 class HealthServicer(_health_pb2_grpc.HealthServicer):
     """Servicer handling RPCs for service statuses."""
 
-    def __init__(self):
+    def __init__(self,
+                 experimental_non_blocking=True,
+                 experimental_thread_pool=None):
         self._lock = threading.RLock()
         self._server_status = {}
-        self._watchers = {}
+        self._on_next_callbacks = {}
+        self.Watch.__func__.experimental_non_blocking = experimental_non_blocking
+        self.Watch.__func__.experimental_thread_pool = experimental_thread_pool
 
-    def _on_close_callback(self, watcher, service):
+    def _on_close_callback(self, on_next_callback, service):
 
         def callback():
             with self._lock:
-                self._watchers[service].remove(watcher)
-            watcher.close()
+                self._on_next_callbacks[service].remove(on_next_callback)
+            on_next_callback(None)
```

Comment:
I know we discussed creating our own sentinel object apart from `None`. I'm not inclined in either direction. Just interested to know how you settled on `None`.

Reply:
This just seemed more straightforward, and since I'm also not inclined in either direction, I just picked the simpler option.